### PR TITLE
clean dom up properly on disposal

### PR DESCRIPTION
**Pull request by whitecolor**
This is possible solution for https://github.com/cyclejs/core/issues/263

### CODE REVIEW

**Comment by whitecolor** [comment]
I found a bug currently in this fix (while using with live with app), can be fixed with adding `endWhen(clean$)`  after `fold`.

```js
.fold(patch, rootElement)
.endWhen(clean$)
```
@TylorS @staltz take a look at at it please, and give you opinion.

File: src/makeDOMDriver.ts [hunk 1]
```diff
@@ -62,8 +63,13 @@ function makeDOMDriver(container: string | Element, options?: DOMDriverOptions):
...
     rootElement$.addListener({next: () => {}, error: () => {}, complete: () => {}});
     /* tslint:enable:no-empty */
-
-    return new DOMSource(rootElement$, runStreamAdapter, [], isolateModule, delegators);
+    const source = new DOMSource(rootElement$, runStreamAdapter, [], isolateModule, delegators);
+    const _dispose = source.dispose;
+    source.dispose = () => {
+      clean$.shamefullySendNext('');
+      _dispose.apply(source);
+    };
```
Comment (staltz) [inline src/makeDOMDriver.ts]: Not nice type of monkey patching with mutation. We can put clean$ as a member of DOMSource.

File: src/makeDOMDriver.ts [hunk 2]
```diff
@@ -62,8 +63,13 @@ function makeDOMDriver(container: string | Element, options?: DOMDriverOptions):
     /* tslint:disable:no-empty */
     rootElement$.addListener({next: () => {}, error: () => {}, complete: () => {}});
     /* tslint:enable:no-empty */
-
-    return new DOMSource(rootElement$, runStreamAdapter, [], isolateModule, delegators);
+    const source = new DOMSource(rootElement$, runStreamAdapter, [], isolateModule, delegators);
```
Comment (staltz) [inline src/makeDOMDriver.ts]: Need to update with the latest on diversity branch because now we are using `MainDOMSource`

**Comment by whitecolor** [comment]
1) Are you ok with name `clean$`? maybe `cleanUp$` or some other variant?
2) Are you ok with `xs.merge(preprocessedVNode$, clean$)` 
3) What about injecting `endWhen(clean$)` before `.drop(1)`?

**Comment by staltz** [comment]
1) `_clean$` just to make it clear it's a framework-internal API.
2) and 3) I think it could be `xs.merge(preprocessedVNode$.endWhen(clean$), clean$)`

**Pull request closed without merging**